homeassistant/config.py

```diff
@@ -1134,7 +1134,7 @@ async def async_process_component_config(  # noqa: C901
             except vol.Invalid as ex:
                 async_log_schema_error(
                     ex,
-                    f"{domain}.{p_name}",
+                    f"{p_name}.{domain}",
```

Comment:
Isn't the dot notation confusing to the user? Do we use that anywhere else for our user facing interface or documentation?

It's not an entity_id, which is a place where we use a dot notation.

Reply:
Hmm, I think I can agree with that. Maybe it would be better to write like this:
```
Invalid config for 'demo' 'light' at integrations/iot_domain.yaml, line 17: required key 'option1' not provided
```

Reply:
The latter is better. Can we deduce what the configuration looks like here? If we can use that in the message that will probably make the most sense to the user. In that case I'd start with mentioning the top level YAML key that they should look for.

If the YAML looks like this:

```yaml
light:
  - platform: demo
```

I'd say:

> Invalid config for 'light' from integration 'demo'

If it looks like this:

```yaml
demo:
  - light:
```

I'd say:

> Invalid config for integration 'demo' in 'light'

Reply:
It will always look like in your first example, your second example is handled by integrations on their own as it's not a platform setup managed by config.py.

Reply:
Good! Then your last suggestion is perfect.